snapcraft/plugins/nodejs.py

```diff
@@ -72,8 +74,12 @@ def schema(cls):
 
     def __init__(self, name, options):
         super().__init__(name, options)
-        self._nodejs_tar = sources.Tar(_get_nodejs_release(),
-                                       os.path.join(self.partdir, 'npm'))
+        try:
+            self._nodejs_tar = sources.Tar(
+                    _get_nodejs_release(self.options.node_engine),
+                    os.path.join(self.partdir, 'npm'))
+        except AttributeError:
+            raise AttributeError('node-engine not defined in snapcraft.yaml')
```

Comment:
This probably means it needs to be required in the schema instead of deleting the `required` key on line 71.